packages/vhd-lib/src/vhd/abstractVhd.js

```diff
@@ -0,0 +1,100 @@
+import { computeBatSize } from './_utils'
+
+export class AbstractVhd {
+  header
+  footer
+  blockTable
+  fullBlockSize
+  bitmapSize
+  /**
+   * instantiate a Vhd
+   *
+   * @returns {AbstractVhd}
+   */
+  static async open(handler, path) {
+    throw new Error(`open is not implemented`)
+  }
```

Comment:
Not sure it's really necessary, also the params may depend on the implementations.